--- a/common/inc/nv-mm.h
+++ b/common/inc/nv-mm.h
@@ -207,7 +207,11 @@ static inline void nv_vma_flags_set_word(struct vm_area_struct *vma, unsigned long flags)
 #if defined(NV_VMA_FLAGS_SET_WORD_PRESENT)
     vma_flags_set_word(&vma->flags, flags);
 #else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+    vm_flags_reset(vma, vma->vm_flags | flags);
+#else
     ACCESS_PRIVATE(vma, __vm_flags) |= flags;
+#endif
 #endif
 }
 
@@ -217,7 +221,11 @@ static inline void nv_vma_flags_clear_word(struct vm_area_struct *vma, unsigned long flags)
 #if defined(NV_VMA_FLAGS_SET_WORD_PRESENT)
     vma_flags_clear_word(&vma->flags, flags);
 #else
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+    vm_flags_reset(vma, vma->vm_flags & ~flags);
+#else
     ACCESS_PRIVATE(vma, __vm_flags) &= ~flags;
+#endif
 #endif
 }
 
